extensions/wikia/EmbeddableDiscussions/scripts/ext.embeddableDiscussions.js

```diff
@@ -93,7 +96,18 @@ require([
 
 	function performRequest($elem) {
 		var requestUrl = getBaseUrl() + $elem.attr('data-requestUrl'),
-			requestData = JSON.parse($elem.attr('data-requestData'));
+			requestData = JSON.parse($elem.attr('data-requestData')),
+			columnsDetailsClass;
+
+		// Inject proper class for 2 columns display
+		if (requestData.columns === 2) {
+			if ($elem.closest('.main-page-tag-rcs').length) {
+				// When the tag is inside the right rail
+				columnsDetailsClass = 'embeddable-discussions-post-detail-right-rail';
```

Comment:
I'm also not certain `right-rail` is the right name here, since that's the name we use for the non-user controlled right rail on article pages, so for clarity, maybe `right-column` since it's referring to the main page right column tag controlled by users?